tests/Unit/DataStructures/Test_DataVector.cpp

```diff
@@ -321,6 +322,47 @@ SPECTRE_TEST_CASE("Unit.DataStructures.DataVector.Math",
   x = DataVector(num_pts, 2.);
   check_vectors(DataVector(num_pts, 0.82682181043180603), square(sin(x)));
   check_vectors(DataVector(num_pts, -0.072067555747765299), cube(cos(x)));
+
+  // Test addition of arrays of DataVectors to arrays of doubles.
+  const DataVector t1{0.0, 1.0, 2.0, 3.0};
+  const DataVector t2{-0.1, -0.2, -0.3, -0.4};
+  const DataVector t3{5.0, 4.0, 3.0, 2.0};
+  const DataVector e1{10.0, 11.0, 12.0, 13.0};
+  const DataVector e2{19.9, 19.8, 19.7, 19.6};
+  const DataVector e3{35.0, 34.0, 33.0, 32.0};
+  const DataVector e4{-10.0, -9.0, -8.0, -7.0};
+  const DataVector e5{-20.1, -20.2, -20.3, -20.4};
+  const DataVector e6{-25.0, -26.0, -27.0, -28.0};
+
+  const std::array<double, 3> point{{10.0, 20.0, 30.0}};
+  const std::array<DataVector, 3> points{{t1, t2, t3}};
+  const std::array<DataVector, 3> expected{{e1, e2, e3}};
+  const std::array<DataVector, 3> expected2{{e4, e5, e6}};
+  CHECK(points + point == expected);
+  CHECK(point + points == expected);
+  CHECK(points - point == expected2);
+  CHECK(point - points == -expected2);
+
```

Comment:
also test arrays of  DV+DV DV-DV DV += (both DV and double) DV -= (both DV and double)